src/generate-services.ts

```diff
@@ -351,6 +367,14 @@ export function generateRpcType(ctx: Context, hasStreamingMethods: boolean): Cod
   const maybeContextParam = options.context ? "ctx: Context," : "";
   const maybeAbortSignalParam = options.useAbortSignal ? "abortSignal?: AbortSignal," : "";
   const methods = [[code`request`, code`Uint8Array`, code`Promise<Uint8Array>`]];
+  const additionalMethods = [];
+  const is$TypePresent = options.outputTypeAnnotations || options.outputTypeRegistry;
+  if (options.outputAfterResponse) {
+    additionalMethods.push(code`afterResponse?(response: { ${is$TypePresent ? "$type: string" : ""} }): void;`);
```

Comment:
@stephenh Since the request / response type is truly dynamic, I was hoping to get your thoughts on this call signature. Options I can think of are the following:

```typescript
// OPTION 1:
beforeRequest: (request: {}): void; // or
beforeRequest: (request: { $type: string }): void;

// OPTION 2: using a index signature
beforeRequest: (request: {[property: string]: any)}: void;

// OPTION 3: using any or object 😢 
beforeRequest: (request: object): void;
```

Reply:
Hm, yeah, I dunno, isn't `request: object` kinda normal for these middleware-esque patterns?

I.e. if you want to listen to "any message going in/out", there really isn't a common/base interface for all messages.

To your point, unless the user has the type registry option turned on, then they'll at least have the `$type` key and can do some "if message.$type === ..." checks.

So maybe `request: object` by default, by `request: { $type: string }` if the type registry feature is enabled?

Fwiw I'm not super-close to the nuances of `object` vs. the index signature, so if you've done some WIP coding of these handlers, and the index signature makes the handler code more ergonomic, I'm good with that too!

Reply:
ended up going with:
```typescript
afterResponse?<T extends { [k in keyof T]: unknown }>(response: T): void;
```